### PR TITLE
Update docs to point to new build scripts

### PR DESCRIPTION
**Pull request by nxtn**
See #32991 for background.

### CODE REVIEW

**Review by hoyosjs — COMMENTED** [review]
This is fine, although for inner loop the idea is to try to use the root scripts more. Also, would of help if we add a note stating this will only build the native bits and not perform any crossgen or test/package building as it used to?

File: docs/workflow/building/coreclr/cross-building.md [hunk 1]
```diff
@@ -134,7 +136,7 @@ The following instructions assume you are on a Linux machine such as Ubuntu 14.0
 To build System.Private.CoreLib for Linux, run the following command:
 
 ```
-    lgs@ubuntu ~/git/coreclr/ $ build.sh arm debug verbose
+    lgs@ubuntu ~/git/runtime/ $ ./src/coreclr/build-runtime.sh arm debug verbose
```
Comment (jkoritzinsky) [inline docs/workflow/building/coreclr/cross-building.md]: To build System.Private.CoreLib, you need to use the root build script now. `build-runtime.sh` doesn't build corelib.

**Review by jkoritzinsky — CHANGES_REQUESTED** [review]

File: docs/workflow/building/coreclr/cross-building.md (hunk 1 above)
Comment (NextTurn) [reply docs/workflow/building/coreclr/cross-building.md]: Updated. Is `--subsetCategory CoreClr` if `--subset CoreLib-NativeCoreLib` is specified?

Comment (NextTurn) [reply docs/workflow/building/coreclr/cross-building.md]: Updated. Is `--subsetCategory CoreClr` unnecessary if `--subset CoreLib-NativeCoreLib` is specified?

**Review by jkoritzinsky — APPROVED** [review]

File: docs/workflow/building/coreclr/cross-building.md (hunk 1 above)
Comment (hoyosjs) [reply docs/workflow/building/coreclr/cross-building.md]: As of now, yes. But it's better to say 
```
./build.(cmd/sh) --subsetCategory CoreClr --subset CoreLib-NativeCoreLib
```
Someday Mono could decide that CoreLib or NativeCoreLib are name of some subset they use. By specifying `subsetCategory` you specify the product you are focusing on.

File: docs/workflow/building/coreclr/cross-building.md [hunk 2]
```diff
@@ -134,13 +136,13 @@ The following instructions assume you are on a Linux machine such as Ubuntu 14.0
 To build System.Private.CoreLib for Linux, run the following command:
 
 ```
-    lgs@ubuntu ~/git/coreclr/ $ build.sh arm debug verbose
+    lgs@ubuntu ~/git/runtime/ $ ./build.sh --subsetCategory CoreClr --subset CoreLib-NativeCoreLib --arch arm -c debug -v verbose
```
Comment (ViktorHofer) [inline docs/workflow/building/coreclr/cross-building.md]: You can just use a single - everywhere. Also lowercasing everything works as well.

**Review by ViktorHofer — APPROVED** [review]

File: docs/workflow/building/coreclr/cross-building.md [hunk 3]
```diff
@@ -11,9 +11,13 @@ Install the ARM tools and Windows SDK, as described [here](https://github.com/do
 Cross compiling CoreCLR
 -----------------------
 
-Build using "-arm" as the architecture. For example:
+Build using "arm" as the architecture. For example:
 
-    C:\runtime> build.cmd -arm -debug
+    C:\runtime> .\build.cmd -subsetCategory CoreClr -subset Runtime -arch arm -c debug
```
Comment (jkotas) [inline docs/workflow/building/coreclr/cross-building.md]:
```suggestion
    C:\runtime> build.cmd -subsetCategory CoreClr -subset Runtime -arch arm -c debug
```

It is unusual to have `.\` on Windows.

File: docs/workflow/building/coreclr/cross-building.md [hunk 4]
```diff
@@ -11,9 +11,13 @@ Install the ARM tools and Windows SDK, as described [here](https://github.com/do
...
+
+-or-
+
+    C:\runtime> .\src\coreclr\build-runtime.cmd -arm -debug
```
Comment (jkotas) [inline docs/workflow/building/coreclr/cross-building.md]:
```suggestion
    C:\runtime> src\coreclr\build-runtime.cmd -arm -debug
```

File: docs/workflow/building/coreclr/cross-building.md (hunk 3 above)
Comment (NextTurn) [reply docs/workflow/building/coreclr/cross-building.md]: Well, this is required by PowerShell.

File: docs/workflow/building/coreclr/cross-building.md (hunk 2 above)
Comment (jkotas) [reply docs/workflow/building/coreclr/cross-building.md]: Making these consistent can be a separate PR.

- The existing docs use a mix of `-` and `--`. 
- `-subset help` lists the names as CamelCased. It makes sense for the docs to match it.

File: docs/workflow/building/coreclr/cross-building.md (hunk 3 above)
Comment (jkotas) [reply docs/workflow/building/coreclr/cross-building.md]: We should have the docs consistent. There is one outlier in the existing docs that has build.cmd with `.\`, the rest has it without it.

Prepending .\ is a second nature for Powershell users, not sure whether we need to have it prepended in the docs.

**Pull request merged**